pypdf/filters.py

```diff
@@ -245,10 +245,10 @@ class ASCIIHexDecode:
 
     @staticmethod
     def decode(
-        data: str,
+        data: Union[str, bytes],
         decode_parms: Union[None, ArrayObject, DictionaryObject] = None,
         **kwargs: Any,
-    ) -> str:
+    ) -> bytes:
```

Comment:
Is this a change that might break something for projects using pypdf?

Reply:
This is quite internal to pypdf. I don't think it will affect. More it is more consistent with most decode functions inhere
(not all : should we think about changing the remaining ones ?)

Reply:
> I'm a big fan of consistency.

👍 

> 
> I've searched [how pypdf.filters are used](https://github.com/search?q=pypdf.filters&type=code). https://github.com/dcmvdbekerom/pypdfplot/blob/192a273e46aaae2be7dcecb9606df1b6aba2640a/pypdfplot/classes.py#L110 uses it. They seem to be using PyPDF4, but that makes me hesitate to merge this breaking change.

Unless I'm wrong They are rewritting the procedure to handle binary instead of str so they are aligning to my approach

> 
> It might be rather internal, but `pypdf.filters` and `ASCIIHexDecode` are both public.

We can include some code to accept str and return an STR in such a case : your opinion ?

> I was thinking about renaming those:
> 
>     * `ASCIIHexDecode` -> `ASCIIHexCodec` or `ASCIIHexFilter`

*ASCIIHexDecode* is the name used in the PDF ref I would prefer to keep it : I propose `ASCIIHexDecodeFilter`

> (...)
> We could have a Codec / Filter interface class with the decode/encode methods from which the others inherit. This way we would not need to make the breaking change now, but could still get your improvement into `pypdf` + we could get a bigger consistency change on the way :-)

encoding is of very limited use for pypdf : Let's prepare the structure and raise exception if called

Reply:
> ASCIIHexDecode is the name used in the PDF ref I would prefer to keep it

Ah, right.

> They are rewritting the procedure to handle binary instead of str so they are aligning to my approach

Hm. All other results are PyPDF4 forks / test code of pypdf. So then it seems ok to merge it like this :thinking:

Reply:
Hi, it's been a while since I've worked on `pypdfplot`, but if memory serves indeed I reimplemented the ASCIIHexDecode filter in bytes. If that is what this PR is about, this saves me work cause I can use the filter straight from `pypdf` :).